Source/astcenc_entry.cpp

```diff
@@ -717,9 +717,48 @@ static void compress_image(
 			int y = rem / row_blocks;
 			int x = rem - (y * row_blocks);
 
-			// Decompress
-			fetch_imageblock(decode_mode, image, &pb, bsd, x * block_x, y * block_y, z * block_z, swizzle);
+			// Test if we can apply some basic alpha-scale RDO
+			bool use_full_block = true;
+			if (ctx.config.a_scale_radius != 0 && block_z == 1)
+			{
+				int start_x = x * block_x;
+				int end_x = astc::min(dim_x, start_x + block_x);
+
+				int start_y = y * block_y;
+				int end_y = astc::min(dim_y, start_y + block_y);
+
+				// Do we have any alpha values?
+				use_full_block = false;
+				for (int ay = start_y; ay < end_y; ay++)
+				{
+					for (int ax = start_x; ax < end_x; ax++)
+					{
+						float a_avg = ctx.input_alpha_averages[ay * dim_x + ax];
+						// SATs accumulate error, so don't test exactly zero
+						// a=0x1 for single pixel in 12x12 block = 2.7e-5 avg
+						if (a_avg > 2.6e-5f)
```

Comment:
This isn't quite true. We need to watch for contribution from pixels outside the block, so the total area that should be considered is `block_size + 2 * alpha radius`.